src/server/components/dashboard/annualReview/helpers.ts

```diff
@@ -22,57 +22,59 @@ function addSixMonths(date: number | string | Date): Date {
   return newDate;
 }
 
-export function getAnnualReviewYear({
-  day,
-  month,
-  lastAnnualReview,
+export function calculateValidDate({
+  userEnteredDate,
+  maxDate,
 }: {
-  day?: string;
-  month?: string;
-  lastAnnualReview: number;
-}): number {
-  const date = new Date(lastAnnualReview);
+  userEnteredDate: Date;
+  maxDate: Date;
+}): Date | null {
+  const userEnteredDateNextYear = DateFns.add(userEnteredDate, { years: 1 });
+  const possibleDates = [userEnteredDate, userEnteredDateNextYear];
 
-  if (!day || !month) return date.getFullYear();
+  const [validDate] = possibleDates.filter(
+    (possibleDate) => DateFns.isBefore(possibleDate, maxDate) && DateFns.isAfter(possibleDate, new Date())
+  );
 
-  const userEnteredDate = new Date(date.getFullYear(), Number(month) + 1, Number(day));
-
-  return DateFns.isBefore(userEnteredDate, date) ? date.getFullYear() + 1 : date.getFullYear();
+  return validDate;
 }
 
-export function getAnnualReviewDate({ day, month, list }: { day: string; month: string; list: List }): {
+export function getAnnualReviewDate({ day, month }: { day: string; month: string }): {
   value: Date | null;
   errorMsg: string | null;
 } {
-  const lastAnnualReview = list.jsonData.lastAnnualReviewStartDate ?? list.createdAt;
-  const annualReviewYear = getAnnualReviewYear({
-    day,
-    month,
-    lastAnnualReview: (list.jsonData.annualReviewStartDate ?? lastAnnualReview) as number,
-  });
-  const parsedDate = DateFns.parse(`${month}/${day}/${annualReviewYear}`, "P", new Date());
   const maxDate = getMaxDate();
+
+  if (!maxDate) throw new Error("confirmNewAnnualReviewDate Error: Max date could not be calculated");
```

Comment:
When would this ever not return a maxDate?

Reply:
good point 👍🏿